control-plane/agents/src/bin/ha/node/server.rs

```diff
@@ -454,23 +458,41 @@ fn parse_uri(new_path: &str) -> Result<ParsedUri, SvcError> {
 struct ParsedUri {
     host: String,
     port: u16,
+    xprt: String,
     nqn: String,
 }
 
 impl ParsedUri {
     fn new(uri: Uri) -> Result<ParsedUri, SvcError> {
         let host = uri.host().ok_or(SvcError::InvalidArguments {})?.to_string();
         let port = uri.port().ok_or(SvcError::InvalidArguments {})?.as_u16();
+        // split returning a None indicates we possibly received old style uri scheme nvmf:// for
+        // new path, which implies transport as tcp.
+        let xprt = uri
+            .scheme_str()
+            .ok_or(SvcError::InvalidArguments {})?
+            .split('+')
+            .nth(1)
+            .unwrap_or("tcp")
```

Comment:
Let's take this from some enum or constants?

Reply:
If we use an enum then could simply be enum::TCP